src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -9427,6 +9427,621 @@ static void Refs(ref R r1, scoped ref R r2)
+                //             r0 = ref r1; // 1
+                Diagnostic(ErrorCode.ERR_RefAssignNarrower, "r0 = ref r1").WithArguments("r0", "r1").WithLocation(8, 13),
+                // (12,13): error CS8374: Cannot ref-assign 'r2' to 'r0' because 'r2' has a narrower escape scope than 'r0'.
+                //             r0 = ref r2; // 2
+                Diagnostic(ErrorCode.ERR_RefAssignNarrower, "r0 = ref r2").WithArguments("r0", "r2").WithLocation(12, 13));
+        }
+
+        [Fact]
+        public void LocalScope_11_Foreach_02()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        S s0 = default;
+        foreach (scoped ref S r0 in Enumerable1.Create(ref s0))
+        {
+            scoped ref S r1 = ref s0;
+            r0 = ref r1;
+
+            scoped ref S r2 = ref r0;
+            r0 = ref r2;
+
+            ref S r3 = ref s0;
+            r0 = ref r3;
+
+            ref S r4 = ref r0;
+            r0 = ref r4;
+
+            break;
+        }
+    }
+}
+ref struct S { }
+
+class Enumerable1
+{
+    public static Enumerable1 Create(ref S p) => default;
+    public Enumerator1 GetEnumerator() => default;
+}
+
+class Enumerator1
+{
+    public ref S Current => throw null;
+    public bool MoveNext() => false;
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (9,13): error CS1656: Cannot assign to 'r0' because it is a 'foreach iteration variable'
+                //             r0 = ref r1;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocalCause, "r0").WithArguments("r0", "foreach iteration variable").WithLocation(9, 13),
+                // (12,13): error CS1656: Cannot assign to 'r0' because it is a 'foreach iteration variable'
+                //             r0 = ref r2;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocalCause, "r0").WithArguments("r0", "foreach iteration variable").WithLocation(12, 13),
+                // (15,13): error CS1656: Cannot assign to 'r0' because it is a 'foreach iteration variable'
+                //             r0 = ref r3;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocalCause, "r0").WithArguments("r0", "foreach iteration variable").WithLocation(15, 13),
+                // (18,13): error CS1656: Cannot assign to 'r0' because it is a 'foreach iteration variable'
+                //             r0 = ref r4;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocalCause, "r0").WithArguments("r0", "foreach iteration variable").WithLocation(18, 13)
+                );
+        }
+
+        [Fact]
+        public void LocalScope_12_Foreach_01()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        int i0 = 0;
+        S s0 = new S(ref i0);
+        {
+            int i1 = 1;
+            foreach (S s1 in Enumerable1.Create(ref i1)) {
+                s0 = s1;
```

Comment:
>s0 = s1; [](http://example.com/codeflow?start=16&length=8)

It looks like this should be an error since the implicit scope of `s1` should be smaller than the scope of `s2`. Is this [https://github.com/dotnet/roslyn/issues/64218](https://github.com/dotnet/roslyn/issues/64218). If so, please consider adding a comment linking to the bug.

Reply:
Will address in the next PR